### ISSUE TITLE
Add docstrings for setnames! etc. 

### ISSUE THREAD

**Issue by pdeffebach** [davidavdav/NamedArrays.jl#88]
There are currently no docstrings for a lot of publicly exposed functions in this package. It would be great to add them.

**Comment by davidavdav** [comment]
Yes you are absolutely right, I should take the effort to do this.  Documentation of this package is pretty poor as it is, anyway.

**Comment by cvigilv** [comment]
Need help with this? I can invest a couple of hours converting the README.md into docstrings? I use this package way to much not to help a little with some documentation and examples.

**Comment by davidavdav** [comment]
Yes, I made a start with some documentation, I don't know how that would work.  

If you send a PR that would be appreciated!
